### PR TITLE
✨  Directorv2: adds current product_name in ScheduledData   [part 2 of #3565] 

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Follows from PR #3803. This PR simply gets information about the current product (i.e. ``product_name``) into the ``directorv2`` ``ScheduledData`` so it can be used to determine the context for  ``OSPARC_ENVIRONMENT_``-identifiers substitution.

Due to the large amount of changes, the next PR will introduce the mechanism that resolves ``OSPARC_ENVIRONMENT_``-identifiers depending on the context (i.e. ``user_id``, ``project_uid``, ``product_name``, etc)

### Changes highlights:

- ✨ ``directorv2`` API 2.1.0 → 2.2.0
   -  NEW ``product_name`` required in API (breaks backwards compatibility) 
   - when a service gets scheduled, it also captures the current ``product_name``  in ``ScheduledData``
- ♻️ cleanup  ``test_utils_string_substitution.py``

## Related issue/s

- Implements part of #3565:  changes in director-v2

## How to test
- directorv2 and webserver tests in-place

## Checklist

- [x] ``make version-*``
- [x] ``make openapi.json``
- [x] Unit tests for the changes exist
- [ ] Runs in the swarm

### CODE REVIEW

**Comment by pcrespov** [comment]
> Could you make sure that if `from_service_inspect` does not receive the `product_name` field encoded in the label, this still works. Just double check this.

@GitHK Can you please guide me on how to test this?

**Review by mguidon — APPROVED** [review]
Lgtm

File: services/director-v2/tests/integration/02/test_dynamic_sidecar_nodeports_integration.py [hunk 1]
```diff
@@ -839,6 +841,7 @@ async def _assert_retrieve_completed(
                 ), "TIP: Message missing suggests that the data was never uploaded: look in services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/nodeports.py"
 
 
+@pytest.mark.testit
```
Comment (sanderegg) [inline services/director-v2/tests/integration/02/test_dynamic_sidecar_nodeports_integration.py:844]: forgotten?

**Review by sanderegg — APPROVED** [review]
all good. you have a missing @testit marker.

**Pull request merged**